api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -353,6 +353,36 @@ const (
 	AWSMachinePublicIPs = "hypershift.openshift.io/aws-machine-public-ips"
 )
 
+// +kubebuilder:validation:Enum=ImageRegistry
+type OptionalCapability string
+
+const ImageRegistryCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityImageRegistry)
+
+// capabilities allows disabling optional components at install time.
+// Once set, it cannot be changed.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.disabledCapabilities) || has(self.disabledCapabilities)", message="disabledCapabilities is required once set"
+type Capabilities struct {
+	// disabledCapabilities when specified, sets the cluster version baselineCapabilitySet to None
+	// and sets all additionalEnabledCapabilities BUT the ones supplied in disabledCapabilities.
+	// This effectively disables that capability on the hosted cluster.
+	//
+	// When this is not supplied, the cluster will use the DefaultCapabilitySet defined for the respective
+	// OpenShift version.
+	//
+	// Removing a capability from disabledCapabilities on a running cluster
+	// will effectively enable the capability on the cluster, causing the
+	// referred component to be installed.
+	//
+	// It is not possible to add capabilities to disabledCapabilities. In
+	// other words, it's not possible to disable a capability at runtime.
+	//
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="disabledCapabilities is immutable"
+	// +optional
+	// +openshift:enable:FeatureGate=DisableClusterCapabilities
```

Comment:
should we add min/max?

Reply:
Right, thanks for reminding me about this. For now we only have one valid value: `ImageRegistry`, so should the min/max both be 13 (len(ImageRegistry))? Or are there some sensible defaults we can use instead?